src/BenchmarksServer/Startup.cs

```diff
@@ -4082,6 +4087,107 @@ private static async Task UseMonoRuntimeAsync(string runtimeVersion, string outp
             }
         }
 
+        private static async Task AOT4Mono(string dotnetSdkVersion, string runtimeVersion, string outputFolder)
+        {
+            try
+            {
+                //Download dotnet sdk package
+                var dotnetMonoPath = Path.Combine(_rootTempDir, "dotnet-mono", $"dotnet-sdk-{dotnetSdkVersion}-linux-x64.tar.gz");
+                if (Directory.Exists(Path.GetDirectoryName(dotnetMonoPath)))
+                {
+                    Directory.Delete(Path.GetDirectoryName(dotnetMonoPath));
+                }
+                Directory.CreateDirectory(Path.GetDirectoryName(dotnetMonoPath));
+
+                Log.WriteLine($"Downloading dotnet skd package for mono AOT");
+
+                var found = false;
+                var url = $"https://dotnetcli.azureedge.net/dotnet/Sdk/{dotnetSdkVersion}/dotnet-sdk-{dotnetSdkVersion}-linux-x64.tar.gz";
+
+                if (await DownloadFileAsync(url, dotnetMonoPath, maxRetries: 3, timeout: 60, throwOnError: false))
+                {
+                    found = true;
+                }
+
+                if (!found)
+                {
+                    throw new Exception("dotnet sdk package not found");
+                }
+                else
+                {
+                    var strCmdTar = $"tar -xf dotnet-sdk-{dotnetSdkVersion}-linux-x64.tar.gz";
+                    var resultTar = runBashCmd(strCmdTar, Path.GetDirectoryName(dotnetMonoPath));
+                }
+
+                Log.WriteLine($"Patching local dotnet with mono runtime and extracting llvm");
+
+                var packageName = "Microsoft.NETCore.App.Runtime.Mono.LLVM.AOT.linux-x64".ToLowerInvariant();
+                var runtimePath = Path.Combine(_rootTempDir, "RuntimePackages", $"{packageName}.{runtimeVersion}.nupkg");
+
+                var strCmdGetVer = "./dotnet --list-runtimes | grep -i \"Microsoft.NETCore.App\"";
+                var resultGetVer = runBashCmd(strCmdGetVer, Path.GetDirectoryName(dotnetMonoPath));
+                var MicrosoftNETCoreAppPackageVersion = resultGetVer.Split(' ')[1];
```

Comment:
You should respect the version that was requested by the driver (`runtimeVersion`)

Reply:
This is for AOT only, so it doesn't matter much that this version is not the same as `runtimeVersion`. I get it because the folder named after this version.